web/packages/teleport/src/components/LabelsInput/LabelsInput.tsx

```diff
@@ -160,80 +161,94 @@ export function LabelsInput({
+        </colgroup>
+        {labels.length > 0 && (
+          <thead>
+            <tr>
+              <th scope="col">
+                <LabelContent required>{labelKey.fieldName}</LabelContent>
+              </th>
+              <th scope="col">
+                <LabelContent required>{labelVal.fieldName}</LabelContent>
+              </th>
+            </tr>
+          </thead>
+        )}
+        <tbody>
+          {labels.map((label, index) => {
+            const validationItem: LabelValidationResult | undefined =
+              validationResult.results?.[index];
+            return (
+              <tr key={index}>
+                <td>
+                  <FieldInput
+                    size={inputSize}
+                    rule={
+                      validationItem
+                        ? precomputed(validationItem.name)
+                        : requiredKey
+                    }
+                    autoFocus={autoFocus}
+                    value={label.name}
+                    placeholder={labelKey.placeholder}
+                    mb={0}
+                    onChange={e => handleChange(e, index, 'name')}
+                    readonly={disableBtns}
+                  />
+                </td>
+                <td>
+                  <FieldInput
+                    size={inputSize}
+                    rule={
+                      validationItem
+                        ? precomputed(validationItem.value)
+                        : requiredField('required')
+                    }
+                    value={label.value}
+                    placeholder={labelVal.placeholder}
+                    mb={0}
+                    onChange={e => handleChange(e, index, 'value')}
+                    readonly={disableBtns}
+                  />
+                </td>
+                <td>
+                  {/* Force the trash button container to be the same height as an
+                      input. We can't just set center-align the cell, because the
+                      field can expand when showing a validation error. */}
+                  <Flex
+                    alignItems="center"
+                    height={inputGeometry[inputSize].height}
                   >
-                    <Icons.Trash size="medium" />
-                  </ButtonIcon>
-                </Flex>
-              </Flex>
-            </Box>
-          );
-        })}
-      </Box>
+                    <ButtonIcon
+                      size={buttonIconSize}
+                      title={`Remove ${adjective}`}
+                      onClick={() => removeLabel(index)}
+                      css={`
+                        &:disabled {
+                          opacity: 0.65;
+                          pointer-events: none;
+                        }
+                      `}
+                      disabled={disableBtns}
+                    >
+                      <Icons.Cross
+                        color={theme.colors.text.muted}
```

Comment:
`design/Icon` seems to support `color="text.muted"`, it's just that it also adds this as an attribute on the HTML element and doesn't have proper type. But it works and saves you from having to `useTheme`.